### DISCUSSION TITLE
Choosing the Best OS for .NET Development - Mac vs. Windows Showdown | Dan Does Code

### DISCUSSION THREAD

**Discussion by giscus** [danielmackay/nextjs-personal-blog#126] [Blog Comments]
# Choosing the Best OS for .NET Development - Mac vs. Windows Showdown | Dan Does Code

As a longtime Windows developer, I decided to try a MacBook Air 13" as a travel device, influenced by my other colleagues who mostly prefer MacBooks. To my surprise, the Mac's performance and battery life far surpassed Windows in every test I conducted. This article is a comparison of the two laptops and the results of the tests.

https://www.dandoescode.com/blog/choosing-the-best-os-for-dotnet-development-mac-vs-windows-showdown

**Comment by SamBWagner** [comment]
Damn, what a great blog. Super detailed and interesting to read. I swapped to a macbook due to a co-worker using one on a Node app. I've never looked back

**Reply by danielmackay** [reply]
Cheers @SamBWagner! 🙂

**Comment by GordonBeeming** [comment]
Welcome to the light side 🤩

I'm I young or pre your sample of getting mac's 😅

It's like I was reading my life story above☺️, I can't believe how life changing the switch was, and ultimate lead to me switching to iphone recently after being an android fanboy for years too after windows phone of course

**Reply by danielmackay** [reply]
haha!  Seeing you switch over to a Mac def got me thinking about it seriously.

I wouldn't mind switching my phone over.  However, the family 'CFO' has yet to approve my 'Purchase Please'... 😂

**Comment by bradystroud** [comment]
Thanks Dan - this is great 🙂🙏🏼

I remember when I first started at SSW and was the odd one out with a  Mac. I had the pleasure of fixing the bugs that only happened on Safari.

https://youtu.be/6FwmGLzyRDk

**Comment by tkapa** [comment]
It's an interesting transition, but Mac has honestly been a great booster for Productivity.

**Reply by danielmackay** [reply]
💯

**Comment by OsirisTerje** [comment]
That is pretty interesting!  At least expected the Mac to be more expensive to beat a Windows one, but not. 
Awesome that you posted this - and scary (- looking down at my own PCs and wondering "are they soon garbage ?" )  :-)

**Reply by danielmackay** [reply]
I feel like the MacBook Air's are good value for money.  The Pro's will cost you another $2000 AUD to get a good specced machine.  Sure they are better, but I'm not sure if they're $2000 better. 😉

**Comment by imabusyman** [comment]
Intel mac was crazy good for .net development, but now im on M3, it went reverse for me.  ARM development for x86/x64 on any ide has been tougher.  Good news, it’s fast on vs2022 (parallels) or rider. Code has been great on Mac though.  I tested a lot on xps 15, i9, 64gb and m3 68mb.

**Comment by Calinator444** [comment]
I'd be interested to know how Mac's stack up against other arm based SOCs like a Snapdragon. It's a little unclear how much of the performance benefits of using a Mac are a result using a more efficient instruction set.
